companyweb_base/models/partner_cweb_model.py

```diff
@@ -0,0 +1,527 @@
+        the partner has to be a company and the partner.vat should be BE000000000"""
+        for rec in self:
+            if rec.is_company and rec.vat and rec.vat.startswith("BE"):
+                rec.cweb_show_button_enhance = True
+            else:
+                rec.cweb_show_button_enhance = False
+
+    @api.depends(
+        "cweb_address_enable",
+        "cweb_street",
+        "cweb_zip",
+        "cweb_city",
+        "cweb_country",
+    )
+    def _compute_show_button_address(self):
+        """for the button to be show
+        the partner has have cweb_address enabled and data for the address field"""
+        for rec in self:
+            if (
+                rec.cweb_address_enable
+                and rec.cweb_street
+                and rec.cweb_zip
+                and rec.cweb_city
+                and rec.cweb_country
+            ):
+                rec.cweb_show_button_address = True
+            else:
+                rec.cweb_show_button_address = False
+
+    @api.depends(
+        "cweb_name_enable",
+        "cweb_jur_form_enable",
+        "cweb_address_enable",
+        "cweb_creditLimit_enable",
+        "cweb_startDate_enable",
+        "cweb_endDate_enable",
+        "cweb_score_enable",
+        "cweb_warnings_enable",
+        "cweb_url_enable",
+        "cweb_vat_liable_enable",
+        "cweb_balance_data_enable",
+        "cweb_prefLang_enable",
+        "cweb_companystatus_enable",
+        "cweb_url_report_enable",
+    )
+    def _compute_show_tab(self):
+        """for the tab with data to be shown
+        at least one of the field _enable has to be true"""
+        for rec in self:
+            if (
+                rec.cweb_name_enable
+                or rec.cweb_jur_form_enable
+                or rec.cweb_address_enable
+                or rec.cweb_creditLimit_enable
+                or rec.cweb_startDate_enable
+                or rec.cweb_endDate_enable
+                or rec.cweb_score_enable
+                or rec.cweb_warnings_enable
+                or rec.cweb_url_enable
+                or rec.cweb_vat_liable_enable
+                or rec.cweb_balance_data_enable
+                or rec.cweb_prefLang_enable
+                or rec.cweb_companystatus_enable
+                or rec.cweb_url_report_enable
+            ):
+                rec.cweb_show_tab = True
+            else:
+                rec.cweb_show_tab = False
+
+    def _create_hash(self, login, password, secret):
+        """method used for the API call
+        it generates the 'loginhash' needed by the API"""
+        today = datetime.now().strftime("%Y%m%d")
+        text = (today + login + password + secret).lower()
+        return sha1(text.encode("utf-8")).hexdigest()
+
+    def _get_country(self, country_code):
+        country = self.env["res.country"].search([("code", "=", country_code.upper())])
+        return country
+
+    def _populate_general_cweb(self, cweb_response):
```

Comment:
```suggestion
    def _cweb_populate_general(self, cweb_response):
```

It's a detail, but it may make thing easier if things that belong together have the same prefix.